test/integration/filters/decode_headers_return_stop_all_filter_2.cc

```diff
@@ -0,0 +1,72 @@
+#include <chrono>
+#include <string>
+
+#include "envoy/event/timer.h"
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/http/common/empty_http_filter_config.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/integration/filters/common.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+// A filter returns StopAllIterationAndBuffer for headers. The iteration continues after 1s.
+class DecodeHeadersReturnStopAllFilter2 : public Http::PassThroughFilter {
```

Comment:
Can you just replace this with DecodeHeadersReturnStopAllFilter and either filter config or header value indicating how long the alarm should go?

Reply:
Fixed. Thanks!